test/solc/CommandLineInterface.cpp

```diff
@@ -148,25 +148,24 @@ BOOST_AUTO_TEST_CASE(version)
 
 BOOST_AUTO_TEST_CASE(multiple_input_modes)
 {
-	std::array<std::string, 10> inputModeOptions = {
+	std::array inputModeOptions {
 		"--help",
 		"--license",
 		"--version",
 		"--standard-json",
 		"--link",
 		"--assemble",
 		"--strict-assembly",
-		"--yul",
 		"--import-ast",
 		"--import-asm-json",
 	};
 	std::string expectedMessage =
 		"The following options are mutually exclusive: "
-		"--help, --license, --version, --standard-json, --link, --assemble, --strict-assembly, --yul, --import-ast, --lsp, --import-asm-json. "
+		"--help, --license, --version, --standard-json, --link, --assemble, --strict-assembly, --import-ast, --lsp, --import-asm-json. "
 		"Select at most one.";
 
-	for (std::string const& mode1: inputModeOptions)
-		for (std::string const& mode2: inputModeOptions)
+	for (auto const& mode1: inputModeOptions)
+		for (auto const& mode2: inputModeOptions)
```

Comment:
You're welcome :) but actually there is a reason for it. Further up there is a declaration of `std::array` that used to be `std::array<std::string, 10>`. Now C++ can infer the type and length of arrays via CTAD / deduction guides so you might as well omit the template arguments. But you can't omit only the length, you must omit both if any. So here this means that it is deduced as `std::array<const char*, 9>`, which of course breaks the for loop with `std::string const& mode1`, as the stuff is implicitly converted and suddenly you are left with a cref to a temporary.
Long story short, I could 
- either write type and length explicitly (which seems superfluous to me and more maintenance work if updating needs to be done), perhaps outright dangerous because the compiler won't complain if the length does not match upon removal, it will just default initialize the tail;
- use string literal suffixes which is -IMO- also meh,
- use auto,
- explicitly write `for (char const* mode : options)`, which, I think, is also less nice than the alternatives.
In this case I really think it is clear that these are string like. But yeah, you are of course also entitled to hating it :)